src/TransportManager.cpp

```diff
@@ -46,15 +46,12 @@ namespace cppkin
         {
             {
                 unique_lock<mutex> lock(m_mut);
-                m_cv.wait_until(lock, chrono::system_clock::now() + chrono::seconds(10), [this]{return m_batchReached; });
+                m_cv.wait_until(lock, chrono::system_clock::now() + chrono::seconds(10), [this]{ return m_batchReached; });
             }
             static vector<Span*> retrievedSpans;
             m_spanQueue.consume_all([](Span* span){retrievedSpans.push_back(span);});
             if(retrievedSpans.size() > 0) {
                 m_transport->Submit(retrievedSpans);
-                for (Span *span : retrievedSpans) {
```

Comment:
There is no evidence that Submit receives **ownership** on the **Spans** under the vector. its not a good interface.
maybe std::vector<std::unique_ptr<Span>> instead? otherwise it is not clear why the delete was removed from the caller side.